test/jtreg_test_thread_factory/src/share/classes/Virtual.java

```diff
@@ -23,12 +23,23 @@
 
 import java.util.concurrent.ThreadFactory;
 
+@SuppressWarnings("removal")
 public class Virtual implements ThreadFactory {
 
     static {
-        // This property is used by ProcessTools and some tests
         try {
+            // This property is used by ProcessTools and some tests
             System.setProperty("test.thread.factory", "Virtual");
+            // Temporary workaround until CODETOOLS-7903526 is fixed in jtreg
+            // The jtreg run main() in the ThreadGroup and catch only exceptions in this group.
+            // The virtual threads don't belong to any group and need global handler.
+            Thread.setDefaultUncaughtExceptionHandler((t, e) -> {
+                    if (e instanceof ThreadDeath) {
```

Comment:
`ThreadDeath` has been deprecated for removal since Java 20, so this should no longer be needed.

Reply:
It is still used in tests and we should ignore it like jtreg doing.

Reply:
Shouldn't this code first retrieve the current default exception handler, and then check whether t is a virtual thread, and if so handle the exception as appropriate (not sure System.exit is appropriate ..). Then for non-virtual threads it delegates to the previous default handler.
```
final UncaughtExceptionHandler originalUEH = Thread.getDefaultUncaughtExceptionHandler();
Thread.setDefaultUncaughtExceptionHandler((t, e) -> {
        if (t.isVirtual()) {
            // ...
        } else {
            originalUEH.uncaughtException(t, e);
        }
});
```

Reply:
There shouldn't be an original UHE. jtreg doesn't set it. The goal is to add this handler for all threads, not only virtual. 
Please note, that it is planned to add it only until the similar problem in jtreg is completely fixed. 
I thought to add something like 
Thread.setDefaultUncaughtExceptionHandler((t, e) -> {
  UHE.ecxeptionThrown = e;
}
...
 @Override
    public Thread newThread(Runnable task) {
        return VIRTUAL_TF.newThread(new Runnable() -> {
                task.run();
                if (UHE.ecxeptionThrown != null) {
                         throw new RuntimeException(UHE.ecxeptionThrown);
                }
            );
    }
}

So test actually throws exceptions. It might miss exceptions for threads that finish later than the main thread. but might it is ok.